### PR TITLE
feat: payggy in group chat

### PR DESCRIPTION
**Pull request by vojtechsimetka**
Implements #269

This requires a bit of polishing. I agreed with @sugarpusher that she'll fix this.

TODO:
- [ ] styling
- [x] navigation on the first screen when the user is in private chat (need to use `history(-2)`)
- [x] 3rd party seeing correct text who sent money to who
- [x] could not replicate ~~investigate the zod error when sending funds~~

**Current views**
Screenshot 2023-08-10 at 14 29 30
Screenshot 2023-08-10 at 14 29 16

**Desired design**
Screenshot 2023-08-10 at 14 29 19

ZOD error:
```
VM24:1 ZodError: [
  {
    "code": "invalid_union",
    "unionErrors": [
      {
        "issues": [
          {
            "received": "success",
            "code": "invalid_literal",
            "expected": "pending",
            "path": [
              "type"
            ],
            "message": "Invalid literal value, expected \"pending\""
          },
          {
            "code": "invalid_type",
            "expected": "number",
            "received": "undefined",
            "path": [
              "transaction",
              "timestamp"
            ],
            "message": "Required"
          },
          {
            "code": "invalid_type",
            "expected": "string",
            "received": "undefined",
            "path": [
              "transaction",
              "hash"
            ],
            "message": "Required"
          }
        ],
        "name": "ZodError"
      },
      {
        "issues": [
          {
            "code": "invalid_type",
            "expected": "number",
            "received": "undefined",
            "path": [
              "transaction",
              "timestamp"
            ],
            "message": "Required"
          },
          {
            "code": "invalid_type",
            "expected": "string",
            "received": "undefined",
            "path": [
              "transaction",
              "hash"
            ],
            "message": "Required"
          }
        ],
        "name": "ZodError"
      },
      {
        "issues": [
          {
            "received": "success",
            "code": "invalid_literal",
            "expected": "error",
            "path": [
              "type"
            ],
            "message": "Invalid literal value, expected \"error\""
          },
          {
            "code": "invalid_type",
            "expected": "number",
            "received": "undefined",
            "path": [
              "transaction",
              "timestamp"
            ],
            "message": "Required"
          },
          {
            "code": "invalid_type",
            "expected": "string",
            "received": "undefined",
            "path": [
              "transaction",
              "hash"
            ],
            "message": "Required"
          },
          {
            "code": "invalid_type",
            "expected": "string",
            "received": "undefined",
            "path": [
              "error"
            ],
            "message": "Required"
          }
        ],
        "name": "ZodError"
      }
    ],
    "path": [],
    "message": "Invalid input"
  }
]
    at get error [as error] (index.mjs:537:31)
    at $$self.$$.update (chat.svelte:31:23)
    at init (Component.js:134:5)
    at new Chat (chat.svelte:38:74)
    at createProxiedComponent (svelte-hooks.js?v=39278044:341:9)
    at new ProxyComponent (proxy.js?v=39278044:242:7)
    at new Proxy<Chat> (proxy.js?v=39278044:349:11)
    at construct_svelte_component_dev (dev.js:257:20)
    at create_fragment (chat.svelte:65:34)
    at init (Component.js:138:34)
```

### CODE REVIEW

File: src/lib/objects/payggy/views/selectUser.svelte [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+		<ChevronLeft />
+	</Button>
+	<Button slot="right" variant="icon" on:click={() => history.go(-2)}>
+		<Close />
+	</Button>
+</Header>
+
+<Container gap={24} justify="center" padX={24}>
+	<p class="text-lg">Who would you like to send tokens to?</p>
+	<InputField bind:value={filterText} placeholder="Search froup members..." />
```
Comment (agazso) [inline src/lib/objects/payggy/views/selectUser.svelte:43]:
```suggestion
	<InputField bind:value={filterText} placeholder="Search group members..." />
```

File: src/lib/objects/payggy/views/selectUser.svelte [hunk 2]
```diff
@@ -0,0 +1,108 @@
+<script lang="ts">
```
Comment (agazso) [inline src/lib/objects/payggy/views/selectUser.svelte:1]: In other files the naming convention is to use dashes as separator between words. According to that this file should be called `select-user.svelte` and not `selectUser.svelte`

**Review by agazso — CHANGES_REQUESTED** [review]
One issue I see with this is that the texts on the widget and the detail view mentions 'You received...' even if I was not part of the transaction.

For example, `a` makes a transaction to `b` and I see it as `c`, will look like as if `c` received the transaction.

**Comment by vojtechsimetka** [comment]
@agazso thanks for reporting it, I did not realize. It's implemented now :)

Screenshot 2023-08-11 at 15 31 18

Screenshot 2023-08-11 at 16 03 49

**Comment by sugarpusher** [comment]
I found an issue, the recipient's Payggy bubble shows the sender as the wrong person

**Review by agazso — APPROVED** [review]

**Pull request merged**